docs/DeveloperGuide.adoc

```diff
@@ -114,99 +186,428 @@ NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X
+
+For example, `sort n/ASC d/DESC` applies sorting by name in ascending order, followed by Date in descending order. This applies to any future commands entered.
+
+==== Implementation
+
+The sorting feature is supported by `SpendingComparator`, a custom comparator to facilitate different sort ordering, and implements the following operation:
+
+* `updateSortedSpendingList(comparator)` -- Updates the sorted spending list with a new comparator.
+
+This operation is exposed in the `Model` interface as `Model#updateSortedSpendingList(comparator)`.
+
+At a high level view, `SpendingComparator` and `SortField` interacts in the manner shown below.
+
+.High-level view of package interaction
+image::SortPackageDiagram.png[Sort Package Diagram]
+
+The sequence diagram below demonstrates how the `sort` command is executed:
+
+.Sequence diagram for an example `sort` command
+image::SortSequenceDiagram.png[Sort Sequence Diagram,width=75%]
+
+The following steps explain the sequence diagram:
+
+1. The user enters `sort n/ASC`.
+2. `LogicManager` calls `SpendingBookParser#parseCommand()`.
+3. `SortCommandParser` is created and validates user input, creating a set of `fields`.
+4. `SortCommand` receives `fields` and stores it.
+5. On `execute()`, `fields` are passed into a `SpendingComparator` object and `Model#updateSortedSpendingList(comparator)` is called to refresh the displayed list.
+
+NOTE: The lifeline for `SortCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+
+The following activity diagram summarises what happens when the user uses the `sort` command:
+
+.Activity diagram for an example `sort` command
+image::SortActivityDiagram.png[Sort Activity Diagram,width=75%]
+
+The following steps explain the activity diagram:
+
+1. The user executes the `sort` command.
+2. If there are valid fields from the input, each field is saved into a list and determines sort order for `Model`.
+3. Else, there are two cases.
+.. If there is an invalid field, display an error message for the invalid field.
+.. If there is a missing field, display an error message for the missing field.
+
+==== Design Considerations
+
+[width="90%", cols="1,2,2", options="header",]
+|===
+|Aspect |Alternative 1 |Alternative 2
+
+|How sort is executed
+|Set a new comparator +
+Pros: Easy to implement +
+Cons: None
+|Sort the underlying list without a comparator +
+Pros: Able to manipulate objects directly +
+Cons: Poor abstraction
+
+|Method design for sort
+|Create filtered list from sorted list +
+Pros: Easy to implement +
+Cons: Minor structure changes
+|Create sorted list from filtered list +
+Pros: Able to test large sorting side effects +
+Cons: Large code structure changes
+
+|===
+
+// tag::reminder[]
+=== Reminder feature
+
+With this reminder feature, users can set the reminders of their tasks with deadlines and delete them when completed.
+
+==== Implementation
```

Comment:
We think it would be most helpful if we could know how a reminder will remind the user.